src/components/account-creation/resend-mfa-code/resend-mfa-code-controller.ts

```diff
@@ -54,7 +54,23 @@ export const resendMfaCodePost = (
 ): ExpressRouteFunc => {
   return async function (req: Request, res: Response) {
     const { sessionId, clientSessionId, persistentSessionId } = res.locals;
-    const { email, phoneNumber } = req.session.user;
+    const {
+      email,
+      isAccountRecoveryJourney,
+      isAccountRecoveryPermitted,
+      phoneNumber,
+    } = req.session.user;
+
+    const isAccountRecoveryEnabledForEnvironment = supportAccountRecovery();
+
+    const accountRecovery =
+      isAccountRecoveryJourney &&
```

Comment:
A question here because I'm not super familiar with this bit of the domain - is `isAccountRecoveryJourney` enough here? Presumably we'd only set it if account recovery was permitted and enabled for the environment?

Reply:
All of the code from this is basically copied from the enterPhoneNumberPost() controller. There this is the way that we determine what journey type we are in and pass it to the api (both enter-phone and resend-mfa-code) are used in both journeys: account-create and  account recovery
So since I would want the data that is sent to the API to be identical with the one from the  enterPhoneNumberPost controller, I did it the same way here. 

The reason for why I want to make sure the data is the same is because backend blocks users on the 6th resend-code request, and the first request of those 6 is always made by the enter phone, leaving 5 request for the resend-mfa-codes. If the data differs from the call in the enter-phone to the one in the resend-code, the resend-code will have 6 resends available because the backend will consider the requests as separate

Reply:
I guess my thoughts are that we only set `isAccountRecovery` to true here: https://github.com/govuk-one-login/authentication-frontend/blob/main/src/components/account-recovery/check-your-email-security-codes/check-account-recovery-middleware.ts#L11-L12 where we already check whether account recovery is permitted, so I'd argue that line here is redundant. 

Checking whether it's enabled for the environment is technically still relevant so fine to leave in. It is switched on in all environments however, so looks like a flag that we can take out, but yeah while that feature flag still exists makes sense to leave it in here.

Not saying the code is wrong in either place, just that some of this seems redundant as checks to me :-)

Reply:
Indeed, I agree with that